File: java/718.java
class Solution {
    public int findLength(int[] A, int[] B) {
        int l = A.length;
        int[][] m = new int[l+1][l+1];
        int max = 0;
        for(int i = 0;i < l;i++){
            for(int j = 0;j < l;j++){
                if(A[i] == B[j]){
                    m[i+1][j+1] = m[i][j] + 1;
                    max= Math.max(max,m[i+1][j+1]);
                }
            }
        }
        return max;
    }
}

File: python/718.py
class Solution:
    def findLength(self, A: List[int], B: List[int]) -> int:
        l = len(A)
        m = [[0]*(l+1) for i in range(l+1)]
        mx = 0
        for i in range(l):
            for j in range(l):
                if A[i] == B[j]:
                    m[i+1][j+1] = m[i][j] + 1
                    mx = max(m[i+1][j+1],mx)
                # else:
                #     m[i+1][j+1] = max(m[i][j+1],m[i+1][j])
        # print(m)
        return mx